File: Backend/src/Modules/ReservationModule/Features/GetRoomInfo/GetRoomInfoResponse.cs
using src.Modules.ReservationModule.Shared.Dtos;

namespace src.Modules.ReservationModule.Features.GetRoomInfo;

public class GetRoomInfoResponse
{
    public required string RoomId { get; init; } 
    public required string RoomName { get; init; } 
    public required List<DeviceDto> RoomDevices { get; init; }
    public required OpenRulesDto OpenTimes { get; init; } 
    public required List<ReservationDto> ReservationDtos { get; init; } 
}

File: Frontend/web/src/models/getRoomInfoResponse.ts
import {ReservationDto} from "./reservationDto.ts";
import {DeviceDto} from "./deviceDto.ts";
import {OpenRulesDto} from "./openRulesDto.ts";

export interface GetRoomInfoResponse {
    roomId: string;
    roomName: string;
    roomDevices: DeviceDto[];
    openTimes: OpenRulesDto;
    reservationDtos: ReservationDto[];

}











